runner/http.go

```diff
@@ -80,12 +81,32 @@ func (h HTTP) ID() string {
 
 // Run executes a GET request to the Path using the Client
 func (h HTTP) Run() op.Op {
+	// protect from accidental nil reference panics
+	if h.ctx == nil {
+		h.ctx = context.Background()
+	}
+
+	runCtx := h.ctx
+	var runCancelFunc context.CancelFunc
+	if h.Timeout > 0 {
+		runCtx, runCancelFunc = context.WithTimeout(h.ctx, time.Duration(h.Timeout))
+		defer runCancelFunc()
+	}
+
 	startTime := time.Now()
 
-	redactedResponse, err := h.Client.RedactGet(h.Path, h.Redactions)
+	redactedResponse, err := h.Client.RedactGetWithContext(runCtx, h.Path, h.Redactions)
 	result := map[string]any{"response": redactedResponse}
 	if err != nil {
-		op.New(h.ID(), result, op.Fail, err, Params(h), startTime, time.Now())
+		var failureType op.Status
+		if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Can we condense this down into a switch?

```
switch err {
case context.DeadlineExceeded:
...
case context.Canceled:
...
default: 
}
```

Reply:
Good idea! Because the context error types are wrapped by the http package, `errors.Is` is needed, but I updated this code to use a `switch`, which does look cleaner!

Reply:
I also added a comment to the code to capture the fact that timeouts/cancellations are wrapped by the http package, so that we remember this in the future.